applications/migrations/0029_replace_harborchoice_harbors_with_resources.py

```diff
@@ -0,0 +1,141 @@
+# Generated by Django 3.1 on 2021-03-25 08:10
+
+from django.db import migrations, models
+import django.db.models.deletion
+
+
+def copy_harbor_choices(apps, schema_editor):
```

Comment:
I'm not super familiar with the M2M relationships using a through model. Can you quickly explain why do we need this tmp model to copy and restore the data? It is not possible to do something like:

```
    for choice in HarborChoice.objects.all():
        choice.harbor=Harbor.objects.filter(servicemap_id=choice.harbor.servicemap_id).first()
        choice.save(update_fields=["harbor"])
```

Reply:
Because this is done before we have changed the `harbor_id` to the new harbors. So they have conflicting types, the "current" `harbor_id: int` and the "new" `harbor_id: UUID`.

By adding this temporary model, we point the choices to the correct `resources.Harbor`, then we change the actual model accordingly now that the data has been backed up in the temp model, so the table will have a type `uuid` instead of `int`. Then, when the model has been corrected, we migrate the data back and get rid of the old model.

Reply:
Ah yep, thanks! Makes sense now that I look at it again :grin: